### ISSUE TITLE
Problems with run language server

### ISSUE THREAD

**Issue by ghost** [vuejs/language-tools#3269]
Hello everyone,

I have a problem with running language server in Helix editor. Interdimietly when I run Vue file it exit. In Helix log file I have output like below:

```
2023-06-10T11:10:58.248 helix_term::application [WARN] Ignoring a client/registerCapability request because dynamic capability registration is not enabled. Please report this upstream to the language server
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/volar-service-pug/node_modules/volar-service-html/out/index.js:65\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "            ...(0, language_service_1.defineProvide)(exports.injectionKeys.htmlDocument, document => {\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "                                                    ^\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "TypeError: (0 , language_service_1.defineProvide) is not a function\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at /nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/volar-service-pug/node_modules/volar-service-html/out/index.js:65:53\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at Object.templateLanguagePlugin (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/volar-service-pug/out/index.js:43:59)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at plugin (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/@vue/language-service/out/plugins/vue-template.js:15:40)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at createLanguageServicePluginContext (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/@vue/language-server/node_modules/@volar/language-service/out/baseLanguageService.js:160:43)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at Object.createLanguageService (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/@vue/language-server/node_modules/@volar/language-service/out/baseLanguageService.js:43:21)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at Object.getLanguageService (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/@vue/language-server/node_modules/@volar/language-server/out/common/project.js:89:42)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at sendDocumentDiagnostics (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/@vue/language-server/node_modules/@volar/language-server/out/common/workspaces.js:119:41)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "    at async updateDiagnostics (/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/@vue/language-server/node_modules/@volar/language-server/out/common/workspaces.js:98:13)\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "\n"
2023-06-10T11:11:00.655 helix_lsp::transport [ERROR] err <- "Node.js v18.16.0\n"
2023-06-10T11:11:00.682 helix_lsp::transport [ERROR] err: <- StreamClosed
```

I'm working on NixOS so I configure it in `.nix` file like below:

```
        {
          name = "vue";
          language-server = {
            command = "vue-language-server";
            args = ["--stdio"];
          };
          config = {
            "typescript" = {
              tsdk = "${nodeDeps}/lib/node_modules/typescript/lib";
            };
            "environment" = "node";
          };
        }
```

The output from above snippet looks like below (`language.toml`): 

```
[[language]]
name = "vue"

[language.config]
environment = "node"

[language.config.typescript]
tsdk = "/nix/store/r1b289b7m3vjhkfy7sr1w0gpm407xa1f-node-dependencies-js-packages-0.0.1/lib/node_modules/typescript/lib"

[language.language-server]
args = ["--stdio"]
command = "vue-language-server"
```

Is someone have similar problem?

**Comment by uros-5** [comment]
I had this error. I just downgraded TS version to `5.0.2`. Which `@volar/vue-language-server` version are you using? Mine is `v1.6.5`` and it works.
Maybe try to update languages-toml.
```toml
# vuejs
[language-server.volar]
command = "vue-language-server"
args = ["--stdio"]
timeout = 5

[language-server.volar.config.typescript]
tsdk = "./node_modules/typescript/lib"

[[language]]
name = "vue"
scope = "source.vue"
file-types = ["vue"]
roots = ["package.json", ".prettierrc.json"]
language-servers = ["volar"]
injection-regex = "vue"
formatter = {command = "prettier", args = ["--parser", "vue"]}
auto-format = true
```

**Comment by bwitczak** [comment]
@uros-5 After your answer I change LSP for Vue from `@vue/language-server` to ` @volar/vue-language-server` and it works! Thank you very much.

**Comment by rodrigocfd** [comment]
> I had this error. I just downgraded TS version to `5.0.2`.

Apparently TypeScript 5.1 breaks a lot of stuff, #3252 is also related to it.

I downgraded my projects to TypeScript 5.0.4, and it came back to work.

**Comment by Siilwyn** [comment]
For anybody running into this downgrading to 1.7.8 fixes this issue for Helix: `@vue/language-server@1.7.8`

**Comment by Siilwyn** [comment]
This is resolved with the latest versions, helix 23.10 and `@vue/language-server@1.8.22` work perfectly.

**Comment by gegoune** [comment]
I am seeing the same error as in https://github.com/vuejs/language-tools/issues/3269#issuecomment-1620102533 with `1.8.27` and node `v21.6.1`.
